test/helper/sending.bash

```diff
@@ -82,20 +155,50 @@ function _send_email() {
 # chosen. Sending more than one mail at any given point in time with this function
 # is UNDEFINED BEHAVIOR!
 function _send_email_and_get_id() {
-  [[ -v CONTAINER_NAME ]] || return 1
+  # Get the name of the variable that the ID is stored in
+  local ID_ENV_VAR_NAME="${1:?Mail ID must be set for _send_email_and_get_id}"
+  # Get a "reference" to the content of ID_ENV_VAR_NAME so we can manipulate the content
+  local -n ID_ENV_VAR_REF=${ID_ENV_VAR_NAME}
+  # Export the variable so everyone has access
+  # shellcheck disable=SC2163
+  export "${ID_ENV_VAR_NAME}"
+  shift 1
 
-  _wait_for_empty_mail_queue_in_container
-  _send_email "${@}"
-  _wait_for_empty_mail_queue_in_container
-
-  local MAIL_ID
+  local QUEUE_ID MESSAGE_ID
   # The unique ID Postfix (and other services) use may be different in length
   # on different systems (e.g. amd64 (11) vs aarch64 (10)). Hence, we use a
   # range to safely capture it.
-  MAIL_ID=$(_exec_in_container tac /var/log/mail.log              \
-    | grep -E -m 1 'postfix/smtpd.*: [A-Z0-9]+: client=localhost' \
-    | grep -E -o '[A-Z0-9]{9,12}' || true)
+  local QUEUE_ID_REGEX='[A-Z0-9]{9,12}'
+  local MESSAGE_ID_REGEX="[0-9]{14}\\.${QUEUE_ID_REGEX}"
```

Comment:
Context comment, no action required.

---

Curious, I hit up the postfix docs and voila [`enable_long_queue_ids`](https://www.postfix.org/postconf.5.html#enable_long_queue_ids). That should explain the difference (_seems related to the inode number?_), it also notes how it defines both the Queue ID and Message ID generation for both `yes` / `no` (default) values.

So it may not have been AMD64 vs ARM64, but rather filesystems and the inode number being encoded into base 52 (`yes`) or base 16 (`no`)? Or just the inode number itself, since leading 0 is possibly dropped (_so short IDs are 5 chars for time + 8 chars for 32-bit inode number_)

With `enable_long_queue_ids=yes` you always get a `z` to delimit the time and inode encoded values. For the inode number encoded, base 16 vs base 51 (_inode is enocded in base 52 without the z_) on my system is producing 6 chars vs 3 chars. On your ARM64 system that is presumably a bit longer, but I don't know about it being ARM64 specific.

---

**Other references:**
- https://github.com/whyscream/postfix-grok-patterns/issues/115#issuecomment-342432837
- https://github.com/MailScanner/v5/issues/15#issuecomment-242133776
- https://groups.google.com/g/mailing.postfix.users/c/4dZ-stoIm2A/m/2oytnpCb3U4J
  > The queue ID is in part based on the message file's inode number (*).
  > The length of a queue ID therefore depends on the range of inode numbers in the file system.
- https://narkive.com/nthhd8C5.2
  > The current implementation, subject to change, uses the inode number followed by the (sub-second portion of the) time in microseconds.
  > The maximal length is determined by the maximal inode number.
   
   > The 5 hex nibbles of microsecond time precede the hex nibbles of the inode
number.